scripts/preprocess/process_compute_centroids.R

```diff
@@ -0,0 +1,22 @@
+process_compute_centroids <- function(outfile, spdat, county_dict_file) {
+  
+  # compute the centroids (actually, return a point on the surface of each multipolygon)
+  centroids <- rgeos::gPointOnSurface(spdat, byid=TRUE, id=spdat@data$GEOID)
+  
+  # edit the properties in the spdat, removing unnecessary ones and adding
+  # others from county_dict. similar to process_counties.R
```

Comment:
I'm a little confused about "remove unnecessary ones". Are there examples? How do you which data is the authoritative county list, `county_dict_file` or `spdat`?

Reply:
`spdat` (aka `county_boundaries_census_sp`) and `county_dict_file` get merged in this file and in process_counties (with a few lines of code duplication in those two files, now noted in both files, that I don't think is worth effort to reduce further). Therefore the county boundaries, centroids, and state boundaries are all more final/authoritative than either spdat or county_dict_file.

In fact, i'll stop pushing the county and state dictionaries to S3 because we're getting away without using them directly in the viz.yaml.

examples of unnecessary ones are all of the columns from spdat@data not included in c(GEOID, STATE_ABBV, and COUNTY_LONG):
```r
> head(spdat@data)
  STATEFP COUNTYFP COUNTYNS       AFFGEOID GEOID    NAME LSAD      ALAND   AWATER
0      01      005 00161528 0500000US01005 01005 Barbour   06 2291820706 50864677
1      01      023 00161537 0500000US01023 01023 Choctaw   06 2365954971 19059247
2      01      035 00161543 0500000US01035 01035 Conecuh   06 2201896058  6643480
3      01      051 00161551 0500000US01051 01051  Elmore   06 1601876535 99850740
4      01      065 00161558 0500000US01065 01065    Hale   06 1667804583 32525874
5      01      109 00161581 0500000US01109 01109    Pike   06 1740741211  2336975
```